client/client.py

```diff
@@ -57,11 +57,41 @@ def heartbeat(self):
                 
                 self.mode = new_data['mode']
                 self.is_primary = new_data['is_primary']
-                
+                if not self.encoding_hash == new_data['encodings_hash']:
+                    self.get_new_encodings()
                 print("Heartbeat response:")
                 print(new_data)
             except:
                 return
+
+
+
+    def get_new_encodings(self):
+            try:
+                encodings = requests.get("http://" + self.ip + ":" + self.port + "/v1/encodings")
+            except:
+                print("couldnt get pickle :(")
+                return
+            os.remove("encodings.pickle")
+            open('encodings.pickle', 'wb').write(encodings.content)
+            self.encoding_data = pickle.loads(open("encodings.pickle", "rb").read())
+            #Update hash
+            self.encoding_hash = Client.calc_hash()
+            print("Updated encodings file")
```

Comment:
Nice! One consideration though, can **opening** the file ever fail? In that case, maybe it's best **not** to delete the old encodings, and instead just overwrite it on a successful file open (if possible).